src/main/scala/com/github/blemale/scaffeine/Scaffeine.scala

```diff
@@ -166,6 +166,35 @@ case class Scaffeine[K, V](underlying: Caffeine[K, V]) {
     Scaffeine(underlying.expireAfterAccess(duration.toNanos, TimeUnit.NANOSECONDS))
 
   /**
+   * Specifies that each entry should be automatically removed from the cache once a duration has
+   * elapsed after the entry's creation, the most recent replacement of its value, or its last
+   * read.
+   *
+   * @param create the length of time an entry should be automatically removed from the cache after the entry's creation.
+   * @param update the length of time an entry should be automatically removed from the cache after the replacement of it's value.
+   * @param read   the length of time an entry should be automatically removed from the cache after the entry's last read.
+   * @tparam K1 the key type of the expiry.
+   * @tparam V1 the value type of the expiry.
+   * @return this builder instance
+   * @throws IllegalStateException if expiration was already set or used with expiresAfterAccess or expiresAfterWrite.
+   */
+  def expireAfter[K1 <: K, V1 <: V](
+    create: (K1, V1, Long) => Duration,
```

Comment:
I would prefer an API like:
```scala
create: (K1, V1, Duration) => Duration
update: (K1, V1, Instant, Duration) => Duration,
read: (K1, V1, Instant, Duration) => Duration
```
WDYT ?

BTW for more context you can take a look at https://github.com/blemale/scaffeine/issues/18.